packages/blade/src/components/Amount/Amount.tsx

```diff
@@ -0,0 +1,271 @@
+import type { ReactElement } from 'react';
+import type { FontSize } from '../../tokens/global/typography';
+import type { HeadingProps, TextProps, TitleProps } from '../Typography';
+import {
+  amountTextSizes,
+  currencyAbbreviationsMapping,
+  currencyLocaleMapping,
+  currencyPrefixMapping,
+  affixFontSizes,
+} from './amountTokens';
+import { BaseText } from '~components/Typography/BaseText';
+import type { Feedback } from '~tokens/theme/theme';
+import type { BaseTextProps } from '~components/Typography/BaseText/types';
+import BaseBox from '~components/Box/BaseBox';
+import type { TestID } from '~src/_helpers/types';
+import { metaAttribute, MetaConstants } from '~utils';
+
+type Currency = 'INR' | 'MYR';
+
+type AmountProps = {
+  /**
+   * The value to be rendered within the component.
+   *
+   */
+  value: number;
+  /**
+   * Sets the intent of the amount.
+   *
+   * @default undefined
+   */
+  intent?: Exclude<Feedback, 'neutral'> | undefined;
+  /**
+   * Sets the size of the amount.
+   *
+   * @default 'heading'
+   */
+  size?:
+    | `title-${NonNullable<Exclude<TitleProps['size'], 'large'>>}`
+    | `heading-${NonNullable<Exclude<HeadingProps<{ variant: 'regular' }>['size'], 'medium'>>}`
+    | `heading-${NonNullable<
+        Exclude<HeadingProps<{ variant: 'regular' }>['size'], 'medium'>
```

Comment:
because heading-medium is not part of the design. 
https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?node-id=28012-580578&t=jI2b5KYW6UPc0bcD-0